cylc/flow/task_job_mgr.py

```diff
@@ -252,7 +261,41 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
         done_tasks = bad_tasks
 
         for _, itasks in sorted(auth_itasks.items()):
-            platform = itasks[0].platform
+            # Find the first platform where >1 host has not been tried and
+            # found to be unreachable.
+            # If there are no good hosts for a task then the task submit-fails.
+            for itask in itasks:
+                # If there are any hosts left for this platform which we
+                # have not previously failed to contact with a 255 error.
+                if (
+                    set(itask.platform['hosts']) -
+                    self.task_remote_mgr.bad_hosts
+                ):
+                    platform = itask.platform
+                    out_of_hosts = False
+                    break
+                else:
+                    # If there are no hosts left for this platform.
+                    # Set the task state to submit-failed.
+                    itask.waiting_on_job_prep = False
+                    itask.local_job_file_path = None
+                    self._prep_submit_task_job_error(
+                        workflow, itask, '(remote init)', ''
+                    )
+                    self.bad_hosts.difference_update(itask.platform['hosts'])
+                    LOG.critical(TaskRemoteMgmtError(
+                        (
+                            f'{itask.platform["name"]}: initialisation did '
+                            f'not complete because it has no reachable hosts.'
+                            f' Clearing {itask.platform["hosts"]} from '
+                            'bad_hosts'
```

Comment:
This might be confusing as a message to users. If the platform has no reachable hosts, why are we removing its hosts from `bad_hosts`?  Also, `bad_hosts` is a variable name rather than the concept exposed to users. Perhaps we should omit the 2nd sentence or rephrase it to something like *"Resetting the list of hosts for {name} to try again next time"*?.

Reply:
I think that ommiting the second sentence makes more sense. Happy to just let the retry logic be part of Cylc's black box. Not knowing about it is probably less confusing for users.